types/coin.go

```diff
@@ -329,51 +329,22 @@ func (coins Coins) safeAdd(coinsB Coins) Coins {
 		panic("Wrong argument: coins must be sorted")
 	}
 
-	sum := ([]Coin)(nil)
-	indexA, indexB := 0, 0
-	lenA, lenB := len(coins), len(coinsB)
-
-	for {
-		if indexA == lenA {
-			if indexB == lenB {
-				// return nil coins if both sets are empty
-				return sum
-			}
+	allCoins := append(coins, coinsB...)
```

Comment:
We should add a comment explaining why we don't append `coins` above and a `//nolint:gocritic` next to this line for making golangci happy.